airflow/contrib/operators/sqoop_operator.py

```diff
@@ -156,9 +176,18 @@ def execute(self, context):
                 input_lines_terminated_by=self.input_lines_terminated_by,
                 input_optionally_enclosed_by=self.input_optionally_enclosed_by,
                 batch=self.batch,
-                relaxed_isolation=self.relaxed_isolation)
+                relaxed_isolation=self.relaxed_isolation,
+                extra_export_options=self.extra_export_options)
         elif self.cmd_type == 'import':
-            if not self.table:
+            # add create hcatalog table to extra import options if option passed
+            # if new params are added to constructor can pass them in here so don't modify sqoop_hook for each param
+            if self.create_hcatalog_table:
+                self.extra_import_options['create-hcatalog-table'] = ''
+
+            if self.table and self.query:
```

Comment:
must be either or. Sqoop throws an error if try them together, which doesn't make sense if anyone tries to run both of these options together

Reply:
created test case for this

Reply:
Thanks for writing the test